### PR TITLE
Ftrace buffer full improvement

### PR DESCRIPTION
**Pull request by jforissier**
A patch to help with ftrace in case a relatively big buffer is used. The first patch is just a minor unrelated improvement.

### CODE REVIEW

File: mk/config.mk [hunk 1]
```diff
@@ -369,6 +369,19 @@ CFG_TA_GPROF_SUPPORT ?= n
...
+CFG_FTRACE_BUF_SIZE ?= 2048
+
+# How to make room when the function tracing buffer is full?
+# 'shift': shift the previously stored data by the amount needed in order
+#    to always keep the latest logs (slower, especially with big buffer sizes)
+# 'wrap': discard the previous data and start at the beginning of the buffer
+#    again (fast, but can result in a mostly empty buffer)
+# 'stop': stop logging new data
+CFG_FTRACE_BUF_WHEN_FULL ?= shift
+$(call force,CFG_FTRACE_BUF_WHEN_FULL_$(CFG_FTRACE_BUF_WHEN_FULL),y)
```
Comment (jenswi-linaro) [inline mk/config.mk]: We should have a test somewhere that only one of `CFG_FTRACE_BUF_WHEN_FULL_{shift, wrap, stop}` is used

Comment (jforissier) [reply mk/config.mk]: There is https://github.com/OP-TEE/optee_os/pull/3601/files#diff-5fe3f587610e26f68128dd3201a33397R105

Do you mean you'd rather have it checked in `mk/config.mk`? (I would tend to agree...)

Comment (jenswi-linaro) [reply mk/config.mk]: Yes, I'd prefer to have the test in a makefile. If it's short enough I suppose we can have it here else perhaps `core/core.mk`

**Comment by jforissier** [comment]
Update

File: mk/config.mk (hunk 1 above)
Comment (jenswi-linaro) [reply mk/config.mk]: How about checking that:
`$(filter y,$(CFG_FTRACE_BUF_WHEN_FULL_shift) $(CFG_FTRACE_BUF_WHEN_FULL_stop) $(CFG_FTRACE_BUF_WHEN_FULL_stop))` == y

Comment (jforissier) [reply mk/config.mk]: That would somewhat redundant with what I have added (`cfg-check-value`), which also provides a good error message.
Ah, maybe I should prefix those derived variables with an underscore, so they wouldn't even be visible in the generated `$(out-dir)/conf.mk` ? (we do have a few already).

Comment (jenswi-linaro) [reply mk/config.mk]: Sounds good

**Comment by jforissier** [comment]
Update

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**